tests/pytest_translate

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env python3
+#
+# Translate a python test to another language.
+#
+import enum
+import sys
+import os
+import argparse
+import ast
+
+sys.path.insert(0, os.path.join(os.path.dirname(
+    os.path.abspath(__file__)), "translate"))
+
+# Not sure how to solve thie pep8 error:
+#   pytest_translate:17:1: E402 module level import not at top of file
+# this cannot be imported until we add the import path above
+import pythonir
+
+parser = argparse.ArgumentParser(
+    description="Translate python test to another language")
+parser.add_argument("input_file")
+parser.add_argument("output_file")
+
+args = parser.parse_args()
+
+
+def eprint(*args, **kwargs):
+    print(*args, file=sys.stderr, **kwargs)
+
+
+eprint(f"input_file: {args.input_file}")
+eprint(f"output_file: {args.output_file}")
+
+with open(args.input_file, "r") as file:
+    input_source = file.read()
+
+try:
+    input_ast = ast.parse(input_source, args.input_file)
+except SyntaxError as syntaxError:
+    sys.exit(f"ERROR: failed to parse '{args.input_file}': {syntaxError}")
+
+
+class Language(enum.Enum):
+    PYTHON = 0
+    CSHARP = 1
+
+
+def get_fileext_language(filename):
+    if filename.endswith(".py"):
+        return Language.PYTHON
+    if filename.endswith(".cs"):
+        return Language.CSHARP
+    return None
+
+
+output_language = get_fileext_language(args.output_file)
+if not output_language:
+    sys.exit(f"ERROR: unknown file extension from '{args.output_file}'")
+eprint(f"output_language: {output_language}")
+
+
+class IrGenVisitor(ast.NodeVisitor):
```

Comment:
I think the visitor class shouldn't be in the same file as the script code but in an imported module.

Reply:
moved to `ir.py`